packages/dev/core/src/XR/webXRExperienceHelper.ts

```diff
@@ -222,13 +234,23 @@ export class WebXRExperienceHelper implements IDisposable {
      * display the first rig camera's view on the desktop canvas.
      * Please note that this will degrade performance, as it requires another camera render.
      * It is also not recommended to enable this in devices like the quest, as it brings no benefit there.
+     * @param enable enable or disable the spectator mode.
+     * @param options giving XRSpectatorModeOption for specutator camera to setup when the spectator mode is enabled.
      */
-    public enableSpectatorMode(): void {
-        if (!this._spectatorMode) {
+    public enableSpectatorMode(enable: boolean, options ?: XRSpectatorModeOption): void {

```

Comment:
I totally agree with the need to enable and disable, but it would be better to have an external function for this (i.e. `disableSpecatatorMode`) that will take care of this, otherwise it is confusing - the enable function takes an enabled flag.

Reply:
Resolved by a645095ed9f74295a06a0c1a81a38f414f11f216